### PR TITLE
ISPN-8002 ScatteredStreamIteratorTest.waitUntilProcessingResults random failures

### PR DESCRIPTION
**Pull request by rvansa**
https://issues.jboss.org/browse/ISPN-8002

Piggybacking little fix in JGroupsTransport.

### CODE REVIEW

File: core/src/test/java/org/infinispan/stream/DistributedStreamIteratorTest.java [hunk 1]
```diff
@@ -203,17 +203,22 @@ public void waitUntilProcessingResults() throws TimeoutException, InterruptedExc
 
       for (Map.Entry<Integer, Set<Map.Entry<Object, String>>> entry : expected.entrySet()) {
          Integer segment = entry.getKey();
+         Set<Map.Entry<Object, String>> answerForSegment = answer.get(segment);
+         if (answerForSegment == null) {
+            fail(String.format("Segment %d, missing all %s", segment, entry.getValue()));
+            continue;
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/stream/DistributedStreamIteratorTest.java]: :D How about `assertEquals(entry.getValue(), answerForSegment)` instead?

**Comment by danberindei** [comment]
@rvansa I had a request for a better `toString()` in stream commands in the JIRA, would you mind adding those as well?

**Comment by rvansa** [comment]
@danberindei Initially I wanted to send only `<` -> `>` PR :) Updated.

File: core/src/main/java/org/infinispan/stream/impl/StreamResponseCommand.java [hunk 2]
```diff
@@ -81,4 +81,14 @@ public boolean isReturnValueExpected() {
    public boolean canBlock() {
       return true;
    }
+
+   @Override
+   public String toString() {
+      final StringBuilder sb = new StringBuilder("StreamResponseCommand{");
+      sb.append(", id=").append(id);
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/stream/impl/StreamResponseCommand.java]: Extra ", "

**Comment by danberindei** [comment]
Integrated, thanks Radim!

**Pull request closed without merging**